noxfile.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+
+from __future__ import absolute_import
+
+import nox
+import os
+# import shutil
+
+
+BLACK_VERSION = "black==19.3b0"
+BLACK_PATHS = ["docs", "google", "tests", "noxfile.py", "setup.py"]
+
+if os.path.exists("samples"):
+    BLACK_PATHS.append("samples")
+
+
+# @nox.session(python="3.7")
+# def lint(session):
+#     """Run linters.
+#
+#     Returns a failure if the linters find linting errors or sufficiently
+#     serious code quality issues.
+#     """
+#     session.install("flake8", BLACK_VERSION)
+#     session.run("black", "--check", *BLACK_PATHS)
+#     session.run("flake8", "google", "tests")
+
+
+# @nox.session(python="3.6")
+# def blacken(session):
+#     """Run black.
+#
+#     Format code to uniform standard.
+#
+#     This currently uses Python 3.6 due to the automated Kokoro run of synthtool.
+#     That run uses an image that doesn't have 3.6 installed. Before updating this
+#     check the state of the `gcp_ubuntu_config` we use for that Kokoro run.
+#     """
+#     session.install(BLACK_VERSION)
+#     session.run("black", *BLACK_PATHS)
+
+
+# @nox.session(python="3.7")
+# def lint_setup_py(session):
+#     """Verify that setup.py is valid (including RST check)."""
+#     session.install("docutils", "pygments")
+#     session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
+
+
+def default(session):
+    # Install all test dependencies, then install this package in-place.
+    session.install("mock", "pytest", "pytest-cov")
+    session.install("-e", ".")
+
+    # Run py.test against the unit tests.
+    session.run(
+        "py.test",
+        # "--quiet",
+        # "--cov=google.cloud",
+        # "--cov=tests.unit",
+        # "--cov-append",
+        # "--cov-config=.coveragerc",
+        # "--cov-report=",
+        # "--cov-fail-under=0",
+        # os.path.join("tests", "unit"),
+        os.path.join("tests", "spanner_dbapi"),
+        *session.posargs,
+    )
+
+
+@nox.session(python=["2.7", "3.5", "3.6", "3.7", "3.8"])
```

Comment:
Is this the right set of target versions?

Reply:
This is what's used in other Google APIs, e.g. Spanner, Bigtable, Storage etc. I'm assuming it's logical to put the same set in here as well.